agent/provider/src/rules.rs

```diff
@@ -238,6 +245,49 @@ impl RulesManager {
         Ok((rulestore_monitor, keystore_monitor, whitelist_monitor))
     }
 
+    fn remove_dangling_rules(&self) -> Result<()> {
+        let keystore_cert_ids = self.keystore.list_ids();
+
+        let mut rulestore = self.rulestore.config.write().unwrap();
+
+        let deleted_partner_rules = Self::remove_rules_not_matching_any_cert(
+            &mut rulestore.outbound.partner,
+            &keystore_cert_ids,
+        );
+        let deleted_audited_payload_rules = Self::remove_rules_not_matching_any_cert(
+            &mut rulestore.outbound.audited_payload,
+            &keystore_cert_ids,
+        );
+        // without dropping lock self.rulestore.save() will not work
+        drop(rulestore);
```

Comment:
Original author had not done it probably because it would be difficult to name such function and its returned value type.
Fixed (but I am not sure if introducing new struct for return type is better than having explicit `drop`)

Reply:
I moved no-self `remove_rules_not_matching_any_cert` outside of scope and added new one taking `self` to `RulesManager` because I had no idea how to name it.